src/hooks/SEP/useDownloadXLSXSEP.ts

```diff
@@ -1,18 +1,17 @@
-import { useCallback } from 'react';
+import { useCallback, useContext } from 'react';
 
+import {
+  DownloadContext,
+  PREPARE_DOWNLOAD_TYPE,
+} from 'context/DownloadContextProvider';
 export function useDownloadXLSXSEP() {
-  const downloadSEPXLSX = useCallback((sepId, callId) => {
-    const element = document.createElement('a');
-    element.setAttribute('href', `/download/xlsx/sep/${sepId}/call/${callId}`);
-    element.setAttribute('download', 'download');
-
-    element.style.display = 'none';
-    document.body.appendChild(element);
-
-    element.click();
-
-    document.body.removeChild(element);
-  }, []);
+  const { prepareDownload } = useContext(DownloadContext);
+  const downloadSEPXLSX = useCallback(
+    (sepId: number, callId: number, name: string) => {
+      prepareDownload(PREPARE_DOWNLOAD_TYPE.XLSX_SEP, [[sepId, callId]], name);
```

Comment:
I am not really sure why we need here `Array<number[]>` for sending `sepId` and `callId`. As I see the acceptable type is `Array<number | number[]>` and here it is a bit confusing that we send `[[sepId, callId]]` and we can send just `[sepId, callId]`. Otherwise it looks great!

Reply:
`sepId` and `callId` together identify the resource we want to download, separating them would mean we want to download 2 files, not one. In most cases, we send the array of IDs, as we want to download multiple resources, but with SEP you need two different IDs to identify one, so we group them.
`useDownloadXLSXSEP` hides this part and the `generateLink` extracts the IDs for us.

Reply:
Great! Good to know.